### ISSUE TITLE
KeplerLightCurve.to_fits() doesn't work with Lightkurve.read() for stitched data.

### ISSUE THREAD

**Issue by ojhall94** [KeplerGO/lightkurve#884]
The code I've used is as follows, and the error is below:

```python3
import lightkurve
sr = lk.search_lightcurve('KIC 1435467', mission='Kepler', cadence='long')
lc = sr[2:5].download_all().stitch()
lc.to_fits('test_llc.fits', overwrite=True)

reread = lk.read('test_llc.fits')

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Users/Anaconda/anaconda3/lib/python3.8/site-packages/astropy/timeseries/sampled.py in read(self, filename, time_column, time_format, time_scale, format, *args, **kwargs)
    365             # First we try the readers defined for the BinnedTimeSeries class
--> 366             return super().read(filename, format=format, *args, **kwargs)
    367 

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/astropy/table/connect.py in __call__(self, *args, **kwargs)
     59 
---> 60         out = registry.read(cls, *args, **kwargs)
     61 

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/astropy/io/registry.py in read(cls, format, cache, *args, **kwargs)
    519         reader = get_reader(format, cls)
--> 520         data = reader(*args, **kwargs)
    521 

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/lightkurve/io/kepler.py in read_kepler_lightcurve(filename, flux_column, quality_bitmask)
     42     # of masking. Details: https://github.com/astropy/astropy/issues/10119
---> 43     quality_mask = KeplerQualityFlags.create_quality_mask(
     44                                 quality_array=lc['sap_quality'],

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/lightkurve/utils.py in create_quality_mask(cls, quality_array, bitmask)
    100         # The bitmask is applied using the bitwise AND operator
--> 101         quality_mask = (quality_array & bitmask) == 0
    102         # Log the quality masking as info or warning

TypeError: ufunc 'bitwise_and' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-14-99b1bd25d84d> in <module>
      4 lc.to_fits('test_llc.fits', overwrite=True)
      5 
----> 6 reread = lk.read('test_llc.fits')

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/lightkurve/io/read.py in read(path_or_url, **kwargs)
     88 
     89     if filetype == "KeplerLightCurve":
---> 90         return KeplerLightCurve.read(path_or_url, format='kepler', **kwargs)
     91     elif filetype == "TessLightCurve":
     92         return TessLightCurve.read(path_or_url, format='tess', **kwargs)

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/lightkurve/lightcurve.py in read(cls, *args, **kwargs)
   2301         if kwargs.get("format") is None:
   2302             kwargs['format'] = "kepler"
-> 2303         return super().read(*args, **kwargs)
   2304 
   2305     def to_fits(self, path=None, overwrite=False, flux_column_name='FLUX',

/Users/Anaconda/anaconda3/lib/python3.8/site-packages/astropy/timeseries/sampled.py in read(self, filename, time_column, time_format, time_scale, format, *args, **kwargs)
    371 
    372             if time_column is None:
--> 373                 raise ValueError("``time_column`` should be provided since the default Table readers are being used.")
    374 
    375             table = Table.read(filename, format=format, *args, **kwargs)

ValueError: ``time_column`` should be provided since the default Table readers are being used.
```

This error doesn't occur when reading in the saved `.fits` file for a single quarter alone.

I'm using MacOS 10.15.6, Astropy version 4.1rc2 and Lightkurve 2.0b3, installed from the repository.

**Comment by barentsen** [comment]
Thanks @ojhall94!

It looks like there are two issues going on:
1. The call to `lc.stitch()` changes the data type of the quality columns from `int` to `float`.  It should not be doing that!
2. The `utils.create_quality_mask()` function crashes when the quality values that are passed are float.  It could be more flexible by attempting to convert them to `int`, perhaps accompanied by a warning.

I won't be able to fix this right away, but would welcome any PR to address one or both of these issues!

**Comment by barentsen** [comment]
I believe the first issue, i.e. `lc.stitch()` changing the data type from `int` to `float`, is a bug in the `vstack` feature of AstroPy.  I opened a bug report here: https://github.com/astropy/astropy/issues/10958

We will want to implement a workaround for this bug in Lightkurve.

**Comment by barentsen** [comment]
Using the master branch after having merged #895, I confirmed that the following snippet now executes without crashing (i.e. the original issue has been resolved):

```python
import lightkurve as lk
sr = lk.search_lightcurve('KIC 1435467', mission='Kepler', cadence='long')
lc = sr[2:5].download_all().stitch()
lc.to_fits('test_llc.fits', overwrite=True)
reread = lk.read('test_llc.fits')
```

*However* I find that `reread` does not preserve all the columns that originally appeared in `lc`.  I'll open a separate issue for that.

**Comment by barentsen** [comment]
I opened #896 to deal with the newly-identified issue I mentioned in the previous comment.

Thanks for reporting this @ojhall94!
